nebula_hw_interfaces/src/nebula_velodyne_hw_interfaces/velodyne_hw_interface.cpp

```diff
@@ -53,7 +103,10 @@ Status VelodyneHwInterface::set_sensor_configuration(
   std::shared_ptr<const VelodyneSensorConfiguration> sensor_configuration)
 {
   auto snapshot = get_snapshot();
-  auto tree = parse_json(snapshot);
+  if (!snapshot.has_value()) {
+    return Status::HTTP_CONNECTION_ERROR;
```

Comment:
`snapshot` already contains a `snapshot.error()` value containing the status, so please return that directly.